### DISCUSSION TITLE
Are the fields connected to datafetcher nullable?

### DISCUSSION THREAD

**Discussion by joont92** [Netflix/dgs-framework#423] [Q&A] [answered]
In `dgs-examples-kotlin`, sample schema is like below  

```graphql
type Show {    
    id: Int!
    title: String!
    releaseYear: Int
    reviews: [Review]
}
```
In `ShowDatafetcher`, it sets `reviews` as null.  
And then `ReviewsFetcher` fills its value.  

But what if `reviews` are mandatory like below?   
```graphql
type Show {    
    id: Int!
    title: String!
    releaseYear: Int
    reviews: [Review]!
    theater: Theater! # Or what if there is any mandatory field like this?
}
```

Should I set it to nullable? 
Or should I have to fill dummy value?
I think setting nullable makes client confusing.

**Comment by paulbakker** [comment, marked as answer]
In this case I would set the schema type to non-nullable, but make the Kotlin type nullable. 
If you're using codegen, you can use the `kotlinAllFieldsOptional` setting to do so.
